roaring/naive_test.go

```diff
@@ -0,0 +1,305 @@
+// Copyright 2017 Pilosa Corp.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package roaring
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestSortSlice(t *testing.T) {
+	a := []uint64{1, 3, 2, 8, 5, 21, 13}
+	sortSlice(a)
+	if !reflect.DeepEqual(a, []uint64{1, 2, 3, 5, 8, 13, 21}) {
+		t.Fatalf("unexpected sorting: %v", a)
+	}
+}
+
+func TestRemoveSliceDuplicates(t *testing.T) {
+	a := []uint64{2, 3, 2, 1, 2, 5, 8, 5, 13, 3, 2, 5, 144}
+	a = removeSliceDuplicates(a)
+
+	if !reflect.DeepEqual(a, []uint64{1, 2, 3, 5, 8, 13, 144}) {
+		t.Fatalf("unexpected values: %v", a)
+	}
+
+	a = append(a, 21, 8, 3, 3, 5, 5, 1, 34, 21, 21)
+	a = removeSliceDuplicates(a)
+
+	if !reflect.DeepEqual(a, []uint64{1, 2, 3, 5, 8, 13, 21, 34, 144}) {
+		t.Fatalf("unexpected values: %v", a)
+	}
+}
+
+func TestIntersectSlice(t *testing.T) {
+	a := []uint64{1, 4, 9, 5, 24, 13}
+	b := []uint64{2, 1, 9, 5, 12}
+	c := intersectSlice(a, b)
+
+	if !reflect.DeepEqual(c, []uint64{1, 5, 9}) {
+		t.Fatalf("unexpected values: %v", c)
+	}
+	if !reflect.DeepEqual(c, intersectSlice(b, a)) {
+		t.Fatalf("unexpected values: %v", c)
+	}
+	if !reflect.DeepEqual(c, intersectSlice(c, a)) {
+		t.Fatalf("unexpected values: %v", c)
+	}
+}
+
+func TestUnionSlice(t *testing.T) {
+	a := []uint64{1, 4, 9, 5, 24, 13}
+	b := []uint64{2, 1, 9, 5, 12}
+	c := unionSlice(a, b)
+
+	if !reflect.DeepEqual(c, []uint64{1, 2, 4, 5, 9, 12, 13, 24}) {
+		t.Fatalf("unexpected values: %v", c)
+	}
+	if !reflect.DeepEqual(c, unionSlice(b, a)) {
+		t.Fatalf("unexpected values: %v", c)
+	}
+	if !reflect.DeepEqual(c, unionSlice(c, a)) {
+		t.Fatalf("unexpected values: %v", c)
+	}
+}
+
+func TestMaxInSlice(t *testing.T) {
+	a := []uint64{1, 4, 9, 5, 24, 13}
+	v := maxInSlice(a)
+	if uint64(24) != v {
+		t.Fatalf("expected %v, but got %v", uint64(24), v)
+	}
+
+	for i := uint64(1000); i <= uint64(100000); i++ {
+		a = append(a, i)
+		if v = maxInSlice(a); v != i {
+			t.Fatalf("expected %v, but got %v", i, v)
+		}
+	}
+}
+
+func TestDifferenceSlice(t *testing.T) {
+	a := []uint64{1, 4, 9, 5, 24, 13}
+	b := []uint64{2, 1, 9, 5, 12}
```

Comment:
it might be nicer to have the test cases as a table up front that have inputs and expected outputs, just so it's easier to see the relation. something like
```
cases := []struct { a, b, aDiffB, bDiffA []uint64 } {
    {
    a: []uint64{1, 4, 9, 5, 24, 13},
    b: []uint64{2, 1, 9, 5, 12},
    aDiffB: []uint64{4, 13, 24},
    bDiffA: []uint64{2, 12},
    },
    }
```